packages/7715-permission-types/src/types.ts

```diff
@@ -164,13 +164,21 @@ export type Erc20TokenStreamPermission = BasePermission & {
  * @property data.tokenAddress - is the address of the ERC20 token to be streamed per period.
  */
 export type Erc20TokenPeriodicPermission = BasePermission & {
-    type: 'erc20-token-periodic';
-    data: MetaMaskBasePermissionData & {
-      periodAmount: Hex;
-      periodDuration: number;
-      startTime?: number | null;
-      tokenAddress: Hex;
-    };
+  type: 'erc20-token-periodic';
+  data: MetaMaskBasePermissionData & {
+    periodAmount: Hex;
+    periodDuration: number;
+    startTime?: number | null;
+    tokenAddress: Hex;
+  };
+};
+
+/**
+ * A permission to revoke an ERC20 token allowance.
+ */
+export type Erc20TokenRevocationPermission = BasePermission & {
+  type: 'erc20-token-revocation';
+  data: MetaMaskBasePermissionData;
```

Comment:
There is no token or spender info here? is this intended to be like that?

Reply:
Yep - the permission grants the recipient authority to revoke _any_ ERC-20 allowance that the permission granter has set.

We could consider narrowing this, but I reasoned that it is a safe permission to be fairly broad with. We can discuss in our upcoming sync.